### PR TITLE
Summarize should have a compile time error for Identifier as Name

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx/issues/2370.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Summarize.cs [hunk 1]
```diff
@@ -124,9 +124,16 @@ public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DTyp
                 // All args starting at index 1 need to be identifiers or aggregate functions (wrapped in AsNode node).
                 switch (arg)
                 {
-                    case AsNode nameNode:
+                    case AsNode asNode:
+
+                        if (asNode.Left is FirstNameNode || asNode.Left is DottedNameNode)
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Summarize.cs:129]: Double checking that we want to also fail on DottedName node?

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Summarize.cs (hunk 1 above)
Comment (anderson-joyle) [reply src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Summarize.cs]: Turns out we shouldn't fail. Thanks.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by gregli-msft — APPROVED** [review]

**Pull request merged**